go/private/rules/stdlib.bzl

```diff
@@ -16,17 +16,107 @@ load("@io_bazel_rules_go//go/private:providers.bzl",
     "GoStdLib",
 )
 
+STDLIB_GOOS_GOARCH = [
+    ("darwin", "amd64"),
+    ("linux", "386"),
+    ("linux", "amd64"),
+    ("linux", "armv6l"),
+    ("windows", "386"),
+    ("windows", "amd64"),
+    ("freebsd", "386"),
+    ("freebsd", "amd64"),
+    ("linux", "ppc64le"),
+    ("linux", "s390x"),
+]
+
+_STDLIB_BUILD = """
+load("@io_bazel_rules_go//go/private:rules/stdlib.bzl", "stdlib")
+
+stdlib(
+    name = "{name}",
+    goos = "{goos}",
+    goarch = "{goarch}",
+    race = {race},
+    cgo = {cgo},
+    visibility = ["//visibility:public"],
+)
+"""
+
 def _stdlib_impl(ctx):
+  go = ctx.actions.declare_file("bin/go") # TODO: .exe
+  src = ctx.actions.declare_directory("src")
+  pkg = ctx.actions.declare_directory("pkg")
+  root_file = ctx.actions.declare_file("ROOT")
+  files = [root_file, go, src, pkg]
+  goroot = root_file.path[:-(len(root_file.basename)+1)]
```

Comment:
Why not `goroot = root_file.dirname`?

Reply:
Probably no good reason, my guess is it's left over from a time when root_file was a string and not a File